pkg/operator/staticpod/controller/prune/prune_controller.go

```diff
@@ -92,85 +85,61 @@ func getRevisionLimits(operatorSpec *operatorv1.StaticPodOperatorSpec) (int32, i
 	if operatorSpec.SucceededRevisionLimit != 0 {
 		succeededRevisionLimit = operatorSpec.SucceededRevisionLimit
 	}
-	return failedRevisionLimit, succeededRevisionLimit
+	return int(failedRevisionLimit), int(succeededRevisionLimit)
 }
 
-func (c *PruneController) excludedRevisionHistory(ctx context.Context, recorder events.Recorder, failedRevisionLimit, succeededRevisionLimit, abandonedRevisionLimit int32) ([]int, error) {
-	var succeededRevisions, failedRevisions, inProgressRevisions, unknownStatusRevisions, abandonedRevisions []int
-
-	configMaps, err := c.configMapGetter.ConfigMaps(c.targetNamespace).List(ctx, metav1.ListOptions{})
-	if err != nil {
-		return []int{}, err
-	}
-	for _, configMap := range configMaps.Items {
-		if !strings.HasPrefix(configMap.Name, statusConfigMapName) {
-			continue
+// revisionsToKeep approximates the set of revisions to keep: spec.failedRevisionsLimit for failed revisions,
+// spec.succeededRevisionsLimit for succeed revisions (for all nodes). The approximation goes by:
+// - don't prune LatestAvailableRevision and the max(spec.failedRevisionLimit, spec.succeededRevisionLimit) - 1 revisions before it.
+// - don't prune a node's CurrentRevision and the spec.succeededRevisionLimit - 1 revisions before it.
+// - don't prune a node's TargetRevision and the spec.failedRevisionLimit - 1 revisions before it.
+// - don't prune a node's LastFailedRevision and the spec.failedRevisionLimit - 1 revisions before it.
+func (c *PruneController) revisionsToKeep(status *operatorv1.StaticPodOperatorStatus, failedLimit, succeededLimit int) sets.Int32 {
```

Comment:
nit: rename to `calculateRevisionsToKeep` (?) so that it doesn't overlap with `revisionsToKeep` function.

Reply:
have renamed the func to defaultedLimits